test/lib.Result.js

```diff
@@ -132,6 +132,20 @@ describe('Lib', function () {
       });
     });
 
+    it('should return rowsAffected on insert error', function (done) {
+      var result = createResult();
+      var insError = new Error('insert error');
+      var reply = {
+        functionCode: FunctionCode.INSERT,
+        rowsAffected: 3
```

Comment:
A real-life `rowsAffected` can be `[ -3, 1, 1 ]`.
-3 stands for 'unique constraint violated'.
1 stands for successfully inserted value.